packages/core/src/components/HelpDrawer/HelpDrawerToggle.jsx

```diff
@@ -0,0 +1,67 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import uniqueId from 'lodash/uniqueId';
+
+/**
+ * A link that triggers the visibility of a help drawer
+ */
+export class HelpDrawerToggle extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    // Use a unique ID to avoid scenarios where multiple toggles on the same
+    // page trigger the same help drawer, which could result in multiple instances
+    // of the drawer being rendered if these weren't unique:
+    this.id = uniqueId('helpDrawer');
```

Comment:
I think all this `id` business might be specific to the App 3 implementation. Looking at the `.example.js` component, I don't see how multiple toggles changing the same bit of component state would cause the component to render multiple help drawers because there's only one in the render function. And if none of that's necessary, I don't think this `HelpDrawerToggle` component is necessary to create either, since it basically becomes a link. That being said, there's a lot of boilerplate to make this link behave like a button, so maybe it's good to keep around.

This also brings up your question about being able to use this in App 3 after it's released. I think we'll just have to use some sort of adapter for the time being, but I don't think that's a bad thing.

Reply:
I'm ok with keeping the toggle a separate component. I removed the id handling from the toggle - do you think the way it works now would work for App3?

Reply:
Yeah, it should still work 👍Looks great